pywinauto/linux/atspi_objects.py

```diff
@@ -0,0 +1,1225 @@
+        else:
+            raise GErrorException(err_p)
+
+    return wrapper
+
+
+ATSPI_ROLE_COUNT = 126
+
+
+def _find_library(libs_list):
+    """Helper for locating native system libraries from the list"""
+    try:
+        process = subprocess.Popen(["ldconfig", "-p"], stdout=subprocess.PIPE, universal_newlines=True)
+        stdout = process.communicate()
+        lines = stdout[0].split("\n")
+        for lib in libs_list:
+            for line in lines:
+                if lib in line:
+                    lib_path = line.split()[3]
+                    print("Lib located: {0}".format(lib_path))
+                    return lib_path
+
+    except IOError:
+        # ldconfig not installed will use default lib name
+        return libs_list[-1]
+
+
+@six.add_metaclass(Singleton)
+class IATSPI(object):
+
+    """Python wrapper around C functions from ATSPI library"""
+
+    LIB = "libatspi"
+    DEFAULT_LIB_NAME = "libatspi.so"
+
+    def __get_roles(self):
+        control_types = []
+        get_role_name = self.atspi.atspi_role_get_name
+        get_role_name.argtypes = [c_int]
+        get_role_name.restype = c_char_p
+        for i in range(ATSPI_ROLE_COUNT):
+            role = get_role_name(i)
+            if role is not None:
+                role = "".join([part.capitalize() for part in role.decode("utf-8").split()])
+                control_types.append(role)
+        return control_types
+
+    def __init__(self):
+        try:
+            self.atspi = cdll.LoadLibrary(_find_library([self.LIB, self.DEFAULT_LIB_NAME]))
+            self.atspi.atspi_init()
+            if not self.atspi.atspi_is_initialized():
+                raise Exception("Cannot initialize atspi module")
+
+            self._control_types = self.__get_roles()
+            self.known_control_types = {}  # string id: numeric id
+            self.known_control_type_ids = {}  # numeric id: string id
+
+            for type_id, ctrl_type in enumerate(self._control_types):
+                self.known_control_types[ctrl_type] = type_id
+                self.known_control_type_ids[type_id] = ctrl_type
+
+        except Exception:
+            message = "atspi library not installed. Please install at-spi2 library or choose another backend"
+            raise Exception(message)
+
+    def get_iface_func(self, func_name):
+        if hasattr(self.atspi, func_name + "_iface"):
+            return getattr(self.atspi, func_name + "_iface")
+        elif hasattr(self.atspi, func_name):
+            return getattr(self.atspi, func_name)
+        else:
+            print("Warning! method: {} not found in libatspi.".format(func_name))
+            return None
+
+
+class GLIB(object):
+
+    """Python wrapper around C functions from GLib library"""
+
+    LIB12 = "libglib-1.2.so"
+    LIB20 = "libglib-2.0.so"
+    LIB22 = "libglib-2.2.so"
+    DEFAULT_LIB_NAME = "libglib.so"
+    glib = cdll.LoadLibrary(_find_library([LIB12, LIB20, LIB22, DEFAULT_LIB_NAME]))
```

Comment:
I think newer versions should have higher priority: `[LIB22, LIB20, LIB12, DEFAULT_LIB_NAME]`